RpcNep5Tracker/RpcNep5Tracker.cs

```diff
@@ -58,58 +62,96 @@ private void ResetBatch()
             }
         }
 
+        private void RecordTransferHistory(Snapshot snapshot, UInt160 scriptHash, UInt160 from, UInt160 to, BigInteger amount, UInt256 txHash, ref ushort transferIndex)
+        {
+            if (!_shouldTrackHistory) return;
+            if (_recordNullAddressHistory || from != UInt160.Zero)
+            {
+                _transfersSent.Add(new Nep5TransferKey(from,
+                        snapshot.GetHeader(snapshot.Height).Timestamp, scriptHash, transferIndex),
+                    new Nep5Transfer
+                    {
+                        Amount = amount,
+                        UserScriptHash = to,
+                        BlockIndex = snapshot.Height,
+                        TxHash = txHash
+                    });
+            }
+
+            if (_recordNullAddressHistory || to != UInt160.Zero)
+            {
+                _transfersReceived.Add(new Nep5TransferKey(to,
+                        snapshot.GetHeader(snapshot.Height).Timestamp, scriptHash, transferIndex),
+                    new Nep5Transfer
+                    {
+                        Amount = amount,
+                        UserScriptHash = from,
+                        BlockIndex = snapshot.Height,
+                        TxHash = txHash
+                    });
+            }
+            transferIndex++;
+        }
+
         private void HandleNotification(Snapshot snapshot, Transaction transaction, UInt160 scriptHash,
             VM.Types.Array stateItems,
             Dictionary<Nep5BalanceKey, Nep5Balance> nep5BalancesChanged, ref ushort transferIndex)
         {
+            if (stateItems.Count == 0) return;
             // Event name should be encoded as a byte array.
             if (!(stateItems[0] is VM.Types.ByteArray)) return;
             var eventName = Encoding.UTF8.GetString(stateItems[0].GetByteArray());
 
-            // Only care about transfers
+            if (eventName == "mintTokens")
```

Comment:
@erikzhang LX is the only MainNet NEP5 token using this non-standard method I think and they are going to contract migrate their NEP5 token; I could just remove handling this. What do you think? 

Also if a NEP5 token uses contract migrate it is problematic for this plugin. Maybe it could detect contract migrate of NEP5 tokens and migrate all the balances in leveldb, but it is not at all efficient.

Reply:
I think we can remove this first. As for contract migrate, can we detect it? Maybe future token contracts should use dynamic calls to prevent changes to the hash when upgrading the contract.

Reply:
> can we detect it?

We probably could detect it; however, migrating all the balance key space in one block is definitely not a good design and also could be a potential DDOS vector.

> Maybe future token contracts should use dynamic calls to prevent changes to the hash when upgrading the contract.

I think it is a great idea to suggest the use of dynamic calls for a contract that may need upgrade so that the NEP5 contract hash would remain the same if contract logic needs to change. We should recommend this as a best practice for contract owners. However, it would be hard to enforce such a thing.

We should give guidance that NEP5 tokens should never have their hash changed; doing so requires an airdrop of their new token to existing holders, because balances will not be tracked by just migrating storage. Alternatively, we should give guidance to contract owners that if they have a NEP5 contract and think they may need to upgrade it's logic in the future using migrate, they should write dynamic invoke with logic in another contract as you suggest so that they prevent changes to the hash.

Reply:
> I think we can remove this first.

I switched it to a config option, and disabled it by default.

Reply:
> We probably could detect it; however, migrating all the balance key space in one block is definitely not a good design and also could be a potential DDOS vector.

Yes, I also thought about this issue. Maybe we should open an issue separately to discuss the solution to this problem. What I can think of right now is not to do a storage migration when upgrading the contract, but to add a redirect record.